src/main/java/io/github/jelastic/core/repository/ElasticRepository.java

```diff
@@ -320,6 +325,46 @@ public void reAlias(@NotEmpty  String newIndex, @NotEmpty String aliasName) {
         return ElasticUtils.getResponse(multiGetItemResponses, idSearchRequest.getKlass());
     }
 
+    /**
+     * Load data of entire index based on batchSize using ES scroll API
+     * @param <T> Response class Type
+     * @param index Index to be loaded
+     * @param query jelastic query object
+     * @param batchSize Will tell ElasticClient the size of each fetch, should be <= 10000 for better performance
+     * @return List<T> list of all objects in that index
+     */
+    public <T> List<T> loadAll(String index, Query query, Integer batchSize, Class<T> klass) {
```

Comment:
Why is it Integer but not int? Why is it boxed?

Reply:
1. Will be changed to `int`